nixos/modules/services/security/fail2ban.nix

```diff
@@ -31,21 +31,35 @@ in
         description = "Whether to enable the fail2ban service.";
       };
 
+      package = mkOption {
+        default = pkgs.fail2ban;
+        type = types.package;
+        description = "The fail2ban package to use for running the fail2ban service.";
+      };
+
+      packageFirewall = mkOption {
```

Comment:
Do we actually need a `package` option for the firewall here given we already have `networking.firewall.package` option? I would have thought that using `config.networking.firewall.package` in this module would be enough.

Reply:
Without this parameter fail2ban will access the package pkgs.iptables directly. And we need to use pkgs.iptables-compat. Also then use pkgs.nftables (need change actions to nftables-multiport)

Reply:
If the `firewall` module uses `config.networking.firewall.package` then you could make it use `pkgs.iptables-compat`. I'm not sure I follow.

Reply:
To change the fail2ban service modification later, I will need the packageFirewall option.
Example my variant fail2ban (corrections required) - https://github.com/Izorkin/nur-packages/blob/master/modules/services/fail2ban.nix